core/base/topoMap/TopoMap.h

```diff
@@ -0,0 +1,1045 @@
+/// \ingroup base
+/// \class ttk::TopoMap
+/// \author Alexandre Talon <alexandre.talon@lip6.fr>
+/// \date March 2023
+///
+///
+/// This module defines the %TopoMap class that embeds points into 2D. The input
+/// is a distance matrix representing the distances between any pair of points
+/// in high dimension. It computes the 2D coordinates of a projection of the
+/// points which preserves the distances of a minimum spanning tree computed by
+/// Kruskal's algorithm. It also preserves the components Kruskal's algorithm
+/// builds iteratively.
+///
+///
+/// \b Related \b publication: \n
+/// "Topomap: A 0-dimensional homology preserving projection of high-dimensional
+/// data"\n Harish Doraiswamy, Julien Tierny, Paulo J. S. Silva, Luis Gustavo
+/// Nonato, and Claudio Silva\n Proc. of IEEE VIS 2020.\n IEEE Transactions on
+/// Visualization and Computer Graphics 27(2): 561-571, 2020.
+
+/// \sa TopoMap
+
+#pragma once
+
+// ttk common includes
+#include <Debug.h>
+#include <Geometry.h>
+#include <UnionFind.h>
+
+// STL includes
+#include <map>
+#include <queue>
+#include <set>
+#include <sstream>
+#include <utility> // For std::pair
+#include <vector>
+
+// cos and sin computations, so we are cautious on testing equalities between
+// double.
+static double Epsilon{ttk::Geometry::pow(10.0, -FLT_DIG + 2)};
+static double EpsilonDBL{ttk::Geometry::pow(10.0, -DBL_DIG + 2)};
+
+template <typename T>
+static inline T compute_dist2(const T *ptA, const T *ptB) {
+  T dx = ptB[0] - ptA[0], dy = ptB[1] - ptA[1];
+
+  return dx * dx + dy * dy;
+}
+
+template <typename T>
+static inline T compute_dist(const T *ptA, const T *ptB) {
+  T dx = ptB[0] - ptA[0], dy = ptB[1] - ptA[1];
+
+  return sqrt(dx * dx + dy * dy);
+}
+
+template <typename T>
+inline bool are_colinear(const T *pptA, const T *pptB, const T *pptC) {
```

Comment:
please use instead `ttk::Geometry::areVectorsColinear()`

Reply:
The reason I had for this, also the angle function was that they expect 3D points.
Maybe I can create a 2D version of these functions in the Geometry package?

Reply:
Agreed with the 2D version of these calls in the geometry package (maybe `angle2D` and `areVectorsColinear2D()`)

Reply:
New function isTriangleColinear2D.